### PR TITLE
Document new aspects of neovim 0.5

### PR DESCRIPTION
**Pull request by teto**
Very much WIP but I wanted to collect some feedback before spending more time. This explodes the current documentation of the website in several subsections: LSP/treesitter/lua etc
2021-04-09_20-37-neovim

@neovim/website-maintainers

### CODE REVIEW

**Comment by teto** [comment]
to be clearer, I dont say we should take the burden of writing documentation etc but just provide answers to someone visiting the website and who has no idea where to look at. The idea is to list external resources under a stable url we can reference in neovim's :help.

File: doc/general.html [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ---
 layout: default
 title: Documentation
-canonical_url: /doc/
```
Comment (jamessan) [inline doc/general.html:4]: This may need some changes to [CI](https://github.com/neovim/doc/blob/main/ci/doc-index.sh).

Comment (teto) [reply doc/general.html:4]: update DOC_INDEX_PAGE_URL you mean ? do you know why it was set != doc/index.html ?

Comment (jamessan) [reply doc/general.html:4]: No clue.  Just ran into it while going through the CI scripts.

Comment (teto) [reply doc/general.html:4]: ok I think I will merge this MR soon. I will be on the lookout for failures and update the CI repo accordingly (maybe add a README as well).

File: doc/treesitter.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+layout: default
+title: Treesitter documentation
+---
+
+{% include nav.html active='Documentation' %}
+
+<section class="front-section">
+  <div class="container">
+    <h1>Treesitter documentation</h1>
```
Comment (teto) [inline doc/treesitter.html:10]: use either tree-sitter or treesitter

**Comment by justinmk** [comment]
> This explodes the current documentation of the website in several subsections: LSP/treesitter/lua

Not a fan of this, appears to be a solution in search of a problem. The extra docs subpages do not really have content that justifies their existence. All of the exploded content could live on the doc/ page itself.

If the doc/ page gets too busy , it is very likely that some of its content isn't really needed and can be removed.

submenus in the header also are not ideal. The "About" submenu was a compromise (and maybe should be revisited), not something we want to proliferate.

**Pull request merged**